### DISCUSSION TITLE
Issue building freerdp-2.0.0-rc2

### DISCUSSION THREAD

**Discussion by johnosbb** [FreeRDP/FreeRDP#7991] [General]
Hi,

I am trying to build freerdp-2.0.0-rc2 on Windows 10 Pro using Visual Studio 17.2.1
I have gathered all of the dependencies into a directory:
and I configure the build with:

```
:: Create a directory for the dependencies
:: mkdir C:\Users\FreeRDP_Evaluation\deps
:: Copy all of the lib, bin and include directories into this dependencies directory
SET DEPENDENCIES=C:\Users\FreeRDP_Evaluation\deps
SET BASE_DIRECTORY=C:\Users\FreeRDP_Evaluation\RemoteApp\RDPClient
SET BUILD_DIRECTORY=%BASE_DIRECTORY%\RDPClient_Build
SET SOURCE_DIRECTORY=%BASE_DIRECTORY%\FreeRDP-2.0.0-rc2
echo "Removing existing directories in %BUILD_DIRECTORY%"
:: Remove all generated CMake files
cd /d %BUILD_DIRECTORY%
for /F "delims=" %%i in ('dir /b') do (rmdir "%%i" /s/q || del "%%i" /s/q)
cd ..
cmake  -G"Visual Studio 17" -B%BUILD_DIRECTORY%  -H%SOURCE_DIRECTORY% -DCMAKE_PREFIX_PATH=%DEPENDENCIES% -DCMAKE_BUILD_TYPE=Release -DCHANNEL_URBDRC=OFF -DWITH_CAIRO=OFF -DWITH_SWSCALE=OFF -DWITH_FFPMEG=OFF -DWITH_OPENSSL=ON -DWITH_SSE2=ON -DCMAKE_GENERATOR_PLATFORM=x64
cmake --build %BUILD_DIRECTORY%
``` 

When I run this I hit:

```
CMake Error at cmake/FindOpenSSL.cmake:298 (list):
  list GET given empty list
Call Stack (most recent call first):
  cmake/FindFeature.cmake:16 (find_package)
  CMakeLists.txt:827 (find_feature)
```

I have installed openssl at
C:\Program Files\OpenSSL-Win64 (Win64 OpenSSL v3.0.4 via https://slproweb.com), also copied into the C:\Users\FreeRDP_Evaluation\deps directory.

Installed cmake --version = cmake version 3.23.1
Any suggestions on this?
I do not hit this issue on the current head of the master.

**Comment by akallabeth** [comment]
1. Please use recent versions of `FreeRDP`, yours is ancient.
2. Better use `ninja` to build, it is a lot faster than taking the detour over visual studio solutions
3. Ensure your `OpenSSL` has a `bin`, `lib`, `include` subfolder structure, so the dependencies will be found
4. just add the proper path to `CMAKE_PREFIX_PATH=<path1>;<path2>;...` (check if `;` or `:` is required, do not build on windows often)

Anyway, https://github.com/FreeRDP/FreeRDP/wiki/Compilation has a (working) compilation instruction (build all from source, install all to the same target directory)

**Reply by johnosbb** [reply]
Thank you for your replay.
The latest version builds fine on windows with ninja, but unfortunately I need to build the rc2 version as well. I will debug the openssl stuff. I just wanted to check before doing that because I do remember this being an issue a few years ago when I last built rc2 on linux and it had to do with openssl versions.

**Reply by akallabeth** [reply]
Ok, FYI, `OpenSSL3` support was added only recently, so you might need an older OpenSSL build for this then.
